### ISSUE TITLE
`.eh_frame` section emitted for `kernel.o` (debug assertions enabled, `rustc` >= 1.70.0)

### ISSUE THREAD

**Issue by ojeda** [Rust-for-Linux/linux#1012]
```txt
ld.lld: warning <internal>:(.eh_frame) is being placed in ‘.eh_frame’ 
```

Being resolved in Rust's side: https://github.com/rust-lang/rust/pull/112403.
See: https://github.com/Rust-for-Linux/linux/issues/355.

**Comment by ojeda** [comment]
Fixed in Rust 1.72.0.

**Comment by ojeda** [comment]
I tested the current nightly (`rustc 1.72.0-nightly (fe7454bf4 2023-06-19)`), and I confirm it is fixed -- thanks again @nbdd0121!

However, I noticed that this also triggers in the `rust` branch for `samples/rust/rust_echo_server.rs`: https://github.com/Rust-for-Linux/linux/issues/1016. I assume due to something related to the `async` usage, i.e. the one that Wedson noticed back then (and it is still there in the current nightly, unlike this one, since I tested it anyhow).

(I will close this one when we land 1.72.0 in mainline)
